File: src/dotnet/FaxCore.ApiSamples/Samples/User/Message/GetMessageStatus.cs
using System;
using System.Threading.Tasks;

namespace FaxCore.ApiSamples.Samples.User.Message
{
    internal static class GetMessageStatus
    {
        public static Task<string> RunAsync(FaxCoreClient client, SampleConfig config)
        {
            if (string.IsNullOrWhiteSpace(config.MessageId))
            {
                throw new InvalidOperationException("Set FAXCORE_MESSAGE_ID or FaxCoreMessageId before running user.message.status.");
            }

            return client.PostAsync("/api/message/status", new { messageID = config.MessageId });
        }
    }
}

File: src/javascript/user/message/get-message-status.mjs
import { FaxCoreClient, printResult } from '../../common/faxcoreClient.mjs';

const messageID = process.env.FAXCORE_MESSAGE_ID;
if (!messageID) {
  throw new Error('FAXCORE_MESSAGE_ID is required.');
}

const client = new FaxCoreClient();
const result = await client.post('/api/message/status', { messageID });

printResult('Message status', result);

